native_client/swift/deepspeech_ios_test/deepspeech_ios_test/AppDelegate.swift

```diff
@@ -0,0 +1,255 @@
+
+func render(audioContext: AudioContext?, stream: DeepSpeechStream) {
+    guard let audioContext = audioContext else {
+        fatalError("Couldn't create the audioContext")
+    }
+
+    let sampleRange: CountableRange<Int> = 0..<audioContext.totalSamples
+
+    guard let reader = try? AVAssetReader(asset: audioContext.asset)
+        else {
+            fatalError("Couldn't initialize the AVAssetReader")
+    }
+
+    reader.timeRange = CMTimeRange(start: CMTime(value: Int64(sampleRange.lowerBound), timescale: audioContext.asset.duration.timescale),
+                                   duration: CMTime(value: Int64(sampleRange.count), timescale: audioContext.asset.duration.timescale))
+
+    let outputSettingsDict: [String : Any] = [
+        AVFormatIDKey: Int(kAudioFormatLinearPCM),
+        AVLinearPCMBitDepthKey: 16,
+        AVLinearPCMIsBigEndianKey: false,
+        AVLinearPCMIsFloatKey: false,
+        AVLinearPCMIsNonInterleaved: false
+    ]
+
+    let readerOutput = AVAssetReaderTrackOutput(track: audioContext.assetTrack,
+                                                outputSettings: outputSettingsDict)
+    readerOutput.alwaysCopiesSampleData = false
+    reader.add(readerOutput)
+
+    var sampleBuffer = Data()
+
+    // 16-bit samples
+    reader.startReading()
+    defer { reader.cancelReading() }
+
+    while reader.status == .reading {
+        guard let readSampleBuffer = readerOutput.copyNextSampleBuffer(),
+            let readBuffer = CMSampleBufferGetDataBuffer(readSampleBuffer) else {
+                break
+        }
+        // Append audio sample buffer into our current sample buffer
+        var readBufferLength = 0
+        var readBufferPointer: UnsafeMutablePointer<Int8>?
+        CMBlockBufferGetDataPointer(readBuffer,
+                                    atOffset: 0,
+                                    lengthAtOffsetOut: &readBufferLength,
+                                    totalLengthOut: nil,
+                                    dataPointerOut: &readBufferPointer)
+        sampleBuffer.append(UnsafeBufferPointer(start: readBufferPointer, count: readBufferLength))
+        CMSampleBufferInvalidate(readSampleBuffer)
+
+        let totalSamples = sampleBuffer.count / MemoryLayout<Int16>.size
+        print("read \(totalSamples) samples")
+
+        sampleBuffer.withUnsafeBytes { (samples: UnsafeRawBufferPointer) in
+            let unsafeBufferPointer = samples.bindMemory(to: Int16.self)
+            stream.feedAudioContent(buffer: unsafeBufferPointer)
+        }
+
+        sampleBuffer.removeAll()
+    }
+
+    // if (reader.status == AVAssetReaderStatusFailed || reader.status == AVAssetReaderStatusUnknown)
+    guard reader.status == .completed else {
+        fatalError("Couldn't read the audio file")
+    }
+}
+
+func test(model: DeepSpeechModel, audioPath: String, completion: @escaping () -> ()) {
+    let url = URL(fileURLWithPath: audioPath)
+
+    //var format = AudioStreamBasicDescription.init()
```

Comment:
Is this left commented on purpose for debug ?

Reply:
This is an accident, will remove it.